pkg/universe.dagger.io/docker/push.cue

```diff
@@ -20,14 +20,26 @@ import (
 	}
 
 	// Image to push
-	image: #Image
+	image?: #Image
+
+	// Images to push
+	images: [K=string]: #Image
 
 	_push: core.#Push & {
 		"dest": dest
 		if auth != _|_ {
 			"auth": auth
 		}
-		input:  image.rootfs
-		config: image.config
+		if image != _|_ {
+			input:    image.rootfs
+			config:   image.config
+			platform: image.platform
+		}
+		for _p, _image in images {
+			inputs: "\(_p)": {
+				input:  _image.rootfs
+				config: _image.config
+			}
+		}
```

Comment:
Disjunction is better than condition: https://docs.dagger.io/1226/coding-style#favor-disjunctions-over-if-conditions

Reply:
Like this?
```cue

// Upload an image to a remote repository
#Push: {
	// Destination ref
	dest: #Ref

	// Complete ref after pushing (including digest)
	result: #Ref & _push.result

	// Registry authentication
	auth?: {
		username: string
		secret:   dagger.#Secret
	}

	_push: core.#Push & {
		"dest": dest
		if auth != _|_ {
			"auth": auth
		}
	}

	{
		// Image to push
		image: #Image

		_push: {
			input:    image.rootfs
			config:   image.config
			platform: image.platform
		}
	} | {
		// Images to push
		images: [K=string]: #Image

		_push: inputs: {
			for _p, _image in images {
				"\(_p)": {
					input:  _image.rootfs
					config: _image.config
				}
			}
		}
	}
}
```